datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -2849,6 +2866,24 @@ mod tests {
             simplify(in_list(col("c1"), vec![lit(1), lit(2)], true)),
             col("c1").not_eq(lit(2)).and(col("c1").not_eq(lit(1)))
         );
+
+        let subquery = Arc::new(test_table_scan_with_name("test").unwrap());
```

Comment:
Can you please also add a negative test (with two subqueries)?

Reply:
of course